packages/core-typings/src/IUpload.ts

```diff
@@ -20,4 +20,29 @@ export interface IUpload extends IRocketChatRecord {
 			height: number;
 		};
 	};
+	store?: string;
+	path?: string;
+	token?: string;
+	uploadedAt?: Date;
+	modifiedAt?: Date;
+	url?: string;
+	originalStore?: string;
+	originalId?: string;
+	message_id?: string;
+	instanceId?: string;
+	AmazonS3?: {
+		path: string;
+	};
+	s3?: {
+		path: string;
+	};
+	GoogleStorage?: {
+		path: string;
+	};
+	googleCloudStorage?: {
+		path: string;
+	};
+	Webdav?: {
+		path: string;
+	};
```

Comment:
keep in mind: instead of creating a bunch of optional props we can use a specialized types, or augmentations